src/components/TextInput/Label/InputLabel.tsx

```diff
@@ -70,6 +71,16 @@ const InputLabel = (props: InputLabelProps) => {
     ],
   };
 
+  let textColor;
+
+  if (error && errorColor) {
+    textColor = errorColor;
+  } else if (isV3 && parentState.value && mode !== 'outlined') {
+    textColor = activeColor;
+  } else {
+    textColor = placeholderColor;
+  }
+
```

Comment:
I don't love an `if/else` statement where the conditions are different from one another 😬
Is there any way this could be broken up into separate `if/else` checks and treat your last value as the default?

```suggestion
  let textColor = placeholderColor;

  if (error && errorColor) {
    textColor = errorColor;
  }
  if (!error && isV3 && parentState.value && mode !== 'outlined') {
    textColor = activeColor;
  } 

```